dts/bindings/sensor/st,iis2iclx-i2c.yaml

```diff
@@ -18,3 +18,8 @@ properties:
         This pin defaults to active high when produced by the sensor.
         The property value should ensure the flags properly describe
         the signal that is presented to the driver.
+
+    int-pin:
+      type: int
+      required: false
+      description: Device INT pin number (1 or 2)
```

Comment:
What's the defined behavior if this property is absent but interrupts are supposed to be used?  That should be documented, if it can happen.

Also now that there are two I think these properties should be pulled out to `st,iis2iclx-common.yaml` and that included by the bus-specific bindings, as with `st,lis2dh`.

Reply:
> What's the defined behavior if this property is absent but interrupts are supposed to be used? That should be documented, if it can happen.
> 

mmmmmh... this is a good point. Actually I think that there is a bug. If I'm not defining int-pin, which is allowed and should default to 1, the code is not building. This I think because I'm using DT_INST_PROP, while I should first check for DT_INST_NODE_HAS_PROP. If not present I'll default to '1'. Does it make sense to you?

And of course I'll mention that in the binding doc


> Also now that there are two I think these properties should be pulled out to `st,iis2iclx-common.yaml` and that included by the bus-specific bindings, as with `st,lis2dh`.

Why two properties? You mean for I2C and SPI? Well, for lis2dh is quite different as that driver is common across multiple devices. But this case is similar to all the other drivers where there is no 'common' structure across the two bus.

Reply:
The description may be something like that:

    int-pin:
      type: int
      required: false
      description: Device INT pin number (1 or 2)

        This number represent which of the two interrupt pins
        (INT1 or INT2) the line is attached to. This property is not
        mandatory and if not present it defaults to 1.

Reply:
> Why two ? You mean for I2C and SPI?

No, I meant `drdy-gpios` and `int`.  Those are common regardless of bus.

> The description may be something like

Looks fine, but there's a preference to put the `default: 1` in the binding s.  A good reason is always "this is the vendor power-up default".  See https://github.com/zephyrproject-rtos/zephyr/issues/25164#issuecomment-701486128

Reply:
> > The description may be something like
> 
> Looks fine, but there's a preference to put the `default: 1` in the binding s. A good reason is always "this is the vendor power-up default". See [#25164 (comment)](https://github.com/zephyrproject-rtos/zephyr/issues/25164#issuecomment-701486128)

But what is the behaviour of 'default' in this case? If the property is missing in the DT would create it anyway with the specified value?

Reply:
Yes.  That's existing behavior.

Reply:
ok, I see now that in #25164 is explained.

Reply:
So, I'd add the default: 1 in the bindings but I'd leave also the driver modifications (check whether the int-pin property is defined and if not default to 1) just in case the property will be deleted. I'm not sure what was the result of #25164 debate.

Reply:
You shouldn't need to do both.  If we end up removing support for `default` all existing uses will have to be examined and switched to either `required` or verified that the driver does something.  But I won't block it if it makes you more comfortable.

Reply:
On reconsideration: one of the arguments I use in #25164 is that having it in two places is a maintenance problem.  So I'd really rather you rely on `default`.

Reply:
Ok, let's rely on default then

Reply:
I wouldn't put the property as required, because it isn't. If the 'default' will get removed then I'd rather change the driver.